pybase/region/client.py

```diff
@@ -128,6 +129,8 @@ def _send_request(self, rq, lock_timeout=10):
 
         # send and receive the request
         future = self.thread_pool.submit(self.send_and_receive_rpc, my_id, rq, to_send)
+        if _async:
+            return future
```

Comment:
We utilize a threadpool here but immediately block. This would work in gevent world but removes the entire point of the threadpool in normal python execution.